tests/suites/test_suite_aes.function

```diff
@@ -151,6 +151,157 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_CIPHER_MODE_XTS */
+void aes_encrypt_xex( char *hex_key_string, char *hex_iv_string,
+                      char *hex_src_string, char *hex_dst_string )
+{
+    unsigned char key_str[100] = { 0, };
+    unsigned char iv_str[100]  = { 0, };
+    unsigned char src_str[100] = { 0, };
+    unsigned char dst_str[100] = { 0, };
+    unsigned char output[100]  = { 0, };
+    mbedtls_aes_xts_context ctx;
+    int key_len, data_len;
+
+    mbedtls_aes_xts_init( &ctx );
+
+    key_len = unhexify( key_str, hex_key_string );
+    unhexify( iv_str, hex_iv_string );
+    data_len = unhexify( src_str, hex_src_string );
+
+    mbedtls_aes_xts_setkey_enc( &ctx, key_str, key_len * 8 );
+
+    TEST_ASSERT( mbedtls_aes_crypt_xex( &ctx, MBEDTLS_AES_ENCRYPT, data_len, iv_str, src_str, output ) == 0);
+    hexify( dst_str, output, data_len );
+    TEST_ASSERT( strcmp( (char *) dst_str, hex_dst_string ) == 0 );
+
+exit:
+    mbedtls_aes_xts_free( &ctx );
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_CIPHER_MODE_XTS */
+void aes_decrypt_xex( char *hex_key_string, char *hex_iv_string,
+                      char *hex_src_string, char *hex_dst_string )
+{
+    unsigned char key_str[100] = { 0, };
+    unsigned char iv_str[100]  = { 0, };
+    unsigned char src_str[100] = { 0, };
+    unsigned char dst_str[100] = { 0, };
+    unsigned char output[100]  = { 0, };
```

Comment:
This should also be stack friendly, but currently isn't. Change XEX test cases to match XTS test case style.